### PR TITLE
Greatly speed up homebrew module when multiple packages are passed in the `name` key 

### PR DESCRIPTION
**Pull request by UnknownPlatypus**
##### SUMMARY

This is a rather big refactoring pull request but with rather big performance improvements.
**This makes this simple brew task 5-15x faster on my machine** by leveraging brew ability to work with multiple packages at a time.  Only 1 brew call is made to install N packages instead of N brew calls previously, greatly reducing the overall runtime.
```yml
- name: Install & upgrade brew packages
  community.general.homebrew:
    name: "{{ thibaut_brew_packages }}"
    state: latest
    update_homebrew: true
```

**Review commit per commit is highly recommended.** 
The first commit adds a few tests to increase coverage and validate more strictly the outputs. 
Then I tried to keep the commits contained and made sure tests are passing for each one of them.

##### ISSUE TYPE
- Refactoring Pull Request
- Test Pull Request

##### COMPONENT NAME
homebrew

##### ADDITIONAL INFORMATION
Here are the improvement I measured locally working with 15 packages (with package sizes from 2 to 250Mo)

- **It went from ~25s to ~1.5s** for the noop case (15 packages already up to date) 
- **It went from ~35s to ~7s** for the few updates case (5/15 packages updated) 

Any brew call takes ~1s because of the overhead of brew itself, this PR tries to cut most of it by doing batch brew calls instead of brew calls in a loop. The performance difference is even more noticeable for small packages were the actual install is fast compared to the brew overhead.

PS: If it makes sense, I'll be more than welcome to be marked as a maintainer of this module. I've seen you might be looking for one in https://github.com/ansible-collections/community.general/issues/8709#issuecomment-2335050203
PS2: This MR is a followup to #9022 were this optimization idea was first mentioned.

### CODE REVIEW

File: changelogs/fragments/9181-improve-homebrew-module-performance.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - homebrew - greatly speed up homebrew module when multiple packages are passed in the `name` key (https://github.com/ansible-collections/community.general/pull/9181).
```
Comment (felixfontein) [inline changelogs/fragments/9181-improve-homebrew-module-performance.yml:2]:
```suggestion
  - homebrew - greatly speed up homebrew module when multiple packages are passed in the ``name`` option (https://github.com/ansible-collections/community.general/pull/9181).
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution. I've added a first quick comment below.

File: changelogs/fragments/9181-improve-homebrew-module-performance.yml (hunk 1 above)
Comment (russoz) [reply changelogs/fragments/9181-improve-homebrew-module-performance.yml:2]: Also no need to name the module in the long description, the line just started naming it.

**Comment by UnknownPlatypus** [comment]
> You might want to take a look at ModuleHelper (not for this PR, of course):
Interesting, I'll look into it for a future PR. 
@russoz Any recommendation of existing module or PR using this helper i could take inspiration from ?

**Comment by felixfontein** [comment]
I'll merge this PR on this weekend if nobody objects.

**Comment by felixfontein** [comment]
@UnknownPlatypus thanks for your contribution!
@russoz thanks for reviewing!

**Pull request merged**